modules/etl/talend/talend-runtime/src/main/java/com/liferay/talend/avro/ResourceCollectionSchemaInferrer.java

```diff
@@ -96,4 +98,59 @@ public static Schema inferSchemaByResourceFields(
 		return schema;
 	}
 
+	public static Schema inferSchemaByResourceType(
+		ApioApiDocumentation.SupportedClass resourceSupportedClass) {
+
+		List<Property> supportedProperties =
+			resourceSupportedClass.getSupportedProperties();
+
+		int size = supportedProperties.size() + 1;
```

Comment:
I don't think you're reusing `size` anywhere else, right? Then you can set this directly in the constructor.